### PR TITLE
Retry getting results

### PR DESCRIPTION
**Pull request by Threated**

### CODE REVIEW

File: src/logic_ask.rs [hunk 1]
```diff
@@ -116,29 +116,33 @@ async fn handle_via_tasks(req: Request<Incoming>, config: &Config, target: &AppI
...
+            .await
+            .map_err(|e| {
+                warn!("Got error from server: {e}");
+                StatusCode::BAD_GATEWAY
+            })?;
+        trace!("Got beam reply: {resp:#?}");
+
+        match resp.status() {
+            StatusCode::OK => break resp,
+            s if tries > 3 => {
```
Comment (TKussel) [inline src/logic_ask.rs:135]: That linebreak is weird

Comment (TKussel) [reply src/logic_ask.rs:135]: Can you pull the retry limit into a separate variable?

File: src/logic_ask.rs [hunk 2]
```diff
@@ -116,29 +116,33 @@ async fn handle_via_tasks(req: Request<Incoming>, config: &Config, target: &AppI
...
+                warn!("Error fetching reply, got code: {s}. Giving up");
+                return Err(StatusCode::BAD_GATEWAY)?;
+            },
+            s => {
```
Comment (TKussel) [inline src/logic_ask.rs:139]: I think it is a good idea to log the 206 and not hide behind GATEWAY_TIMEOUT. 👍

**Review by TKussel — CHANGES_REQUESTED** [review]

File: src/logic_ask.rs (hunk 1 above)
Comment (Threated) [reply src/logic_ask.rs]: Done

**Review by TKussel — APPROVED** [review]

**Pull request merged**